### PR TITLE
Add ioctl subcommand FIONBIO/FIOASYNC

### PR DESCRIPTION
**Pull request by cqs21**
see issue: #960

### CODE REVIEW

**Comment by tatetian** [comment]
@liqinggd Could you review this PR?

File: kernel/aster-nix/src/net/socket/ip/stream/mod.rs [hunk 1]
```diff
@@ -376,7 +379,32 @@ impl FileLike for StreamSocket {
         self.pollee.poll(mask, poller)
     }
 
+    fn ioctl(&self, cmd: IoctlCmd, arg: usize) -> Result<i32> {
```
Comment (liqinggd) [inline kernel/aster-nix/src/net/socket/ip/stream/mod.rs:382]: I think you can define and implement the `ioctl` method in the `Socket` trait, so all the structs which implement `Socket` can handle the `ioctl`.

**Comment by liqinggd** [comment]
Your latest commit places the ioctls FIONBIO/FIOASYNC in the syscall implementation, which is reasonable. In fact, the implementation here serves the same purpose as the F_SETFL of fcntl, both modifying the flags of the file descriptor.

LGTM

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**